lib/repositories/usecases/GetAllLiquidityInfoUseCase.dart

```diff
@@ -0,0 +1,99 @@
+      print("fetching liquidity pairs info");
+      final tokenPairData = await _graphRepo.queryAllPairsForWalletId(walletId);
+
+      if (tokenPairData.isLeft()) {
+        final data = tokenPairData.getLeft().toNullable();
+        if (data != null) {
+          print("data retrieved: ${data.toString()}");
+          final user = data["user"];
+          print("user retrieved: ${user.toString()}");
+          final liquidityPositions = user["liquidityPositions"];
+          print("liquidityPositions =  ${liquidityPositions.toString()}");
+          List<LiquidityPosition> liquidityPositionsParsed = liquidityPositions
+              .map<LiquidityPosition>((liquidityPosition) =>
+                  LiquidityPosition.fromJson(liquidityPosition))
+              .toList();
+          final List<MyLiquidityItemInfo> myLiquidityItemInfosWithZeroBalance =
+              liquidityPositionsParsed
+                  .map((liquidityPosition) =>
+                      getMyLiquidityItemInfoFromLiquidityPosition(
+                          liquidityPosition))
+                  .toList();
+          final List<MyLiquidityItemInfo> myLiquidityItemInfosNoZeroBalance =
+              myLiquidityItemInfosWithZeroBalance
+                  .where((lpPosition) => lpPosition.lpTokenPairBalance != '0').toList();
+          print(myLiquidityItemInfosNoZeroBalance);
+          return Either.left(Success(myLiquidityItemInfosNoZeroBalance));
+        } else {
+          return Either.left(Success(null));
+        }
+      } else {
+        return Either.right(SubgraphError(
+            "Failed to fetch allLiquidityPairsInfo from Subgraph"));
+      }
+    } catch (e) {
+      var errorMsg = "Error occurred fetching allLiquidityPairsInfo: $e";
+      print(errorMsg);
+      return Either.right(SubgraphError(errorMsg));
+    }
+  }
+
+  MyLiquidityItemInfo getMyLiquidityItemInfoFromLiquidityPosition(
+      LiquidityPosition liquidityPosition) {
+    final String token0Name = liquidityPosition.pair.token0.name;
+    final String token1Name = liquidityPosition.pair.token1.name;
+    final String token0Symbol = liquidityPosition.pair.token0.symbol!;
+    final String token1Symbol = liquidityPosition.pair.token1.symbol!;
+    final String token0Address = liquidityPosition.pair.token0.id;
+    final String token1Address = liquidityPosition.pair.token1.id;
+    final String lpTokenPairAddress = liquidityPosition.pair.id;
+    final Decimal reserve0 = Decimal.parse(liquidityPosition.pair.reserve0);
+    final Decimal reserve1 = Decimal.parse(liquidityPosition.pair.reserve1);
+    final Decimal lpTokenPairBalance =
+        Decimal.parse(liquidityPosition.liquidityTokenBalance);
+    final Decimal lpTokenTotalSupply =
+        Decimal.parse(liquidityPosition.pair.totalSupply!);
+    final Decimal shareOfPool = (lpTokenPairBalance / lpTokenTotalSupply)
+        .toDecimal(
+            scaleOnInfinitePrecision:
+                37); //TODO: How to round irrational numbers?
```

Comment:
tag a ticket in the backlog for this or it will never get done

Reply:
done